static/js/common/upload-addon.js

```diff
@@ -514,6 +518,19 @@
                     )
                   : '',
               );
+            // In addition, if we automatically ticked and disabled the Android
+            // checkbox, the browser won't submit the value. It's fine if
+            // Firefox was also checked, but if not then we'd not submit
+            // anything and the validation would fail server-side, so we
+            // add/remove an hidden input to compensate.
```

Comment:
Let me know if that's not clear. The disabled checkbox is really just for showing the developer that the add-on is going to be marked as compatible with Android, it's not submitted since it's disabled. Unfortunately `readonly` is not supported for checkboxes by design in the spec...

Reply:
it's a hack but I can't think of another way